internal/httpproxy/proxy.go

```diff
@@ -0,0 +1,29 @@
+package httpproxy
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// ProxyConfig definition
+type ProxyConfig map[string]StoreConfig
+
+// Proxy definition
+type Proxy map[string]*store
+
+// NewProxy allocates a new http proxy from its configuration
+func New(cfg ProxyConfig) *Proxy {
+	proxy := Proxy{}
+	for k, v := range cfg {
+		proxy[k] = &store{k, v.Mode, []storeEntry{}}
+	}
+	return &proxy
+}
+
+// Listen for the proxy store/retrieve
+func (proxy *Proxy) Listen(mux *http.ServeMux, prefix string) {
```

Comment:
The method name is confusing. "Listen" suggests that we bind to a TCP port using this method. But you are only registring routes here. So "RegisterRoutes" would be the better name here.